src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/TestEnvironment.cs

```diff
@@ -322,29 +348,59 @@ private void InteractiveLogin(string tenant, ActiveDirectoryServiceSettings aadS
             {
-                //TokenCredentials cred = this.TokenInfo[TokenAudience.Management];
+                try { callerId = this.TokenInfo[TokenAudience.Management].CallerId; } catch { }
+            }
 
+            if (!(string.IsNullOrEmpty(this.SubscriptionId)) && !(this.SubscriptionId.Equals("None", StringComparison.OrdinalIgnoreCase)))
+            {
                 List<SubscriptionInfo> subscriptionList = ListSubscriptions(this.BaseUri.ToString(), this.TokenInfo[TokenAudience.Management]);
-                string subscriptionId = subscriptionList.Where((sub) => sub.SubscriptionId.Equals(this.SubscriptionId, StringComparison.OrdinalIgnoreCase)).FirstOrDefault().SubscriptionId;
-
-                if (string.IsNullOrEmpty(subscriptionId))
+                if (subscriptionList.Any<SubscriptionInfo>())
                 {
-                    StringBuilder sb = new StringBuilder("List of subscriptions retrieved:\r\n");
-                    foreach (SubscriptionInfo subInfo in subscriptionList)
+                    var matchedSubs = subscriptionList.Where((sub) => sub.SubscriptionId.Equals(this.SubscriptionId, StringComparison.OrdinalIgnoreCase));
+                    if(matchedSubs.IsAny<SubscriptionInfo>())
+                    {
+                        matchedSubscriptionId = matchedSubs.FirstOrDefault().SubscriptionId;
+                    }
+                    else
                     {
-                        sb.AppendLine(subInfo.SubscriptionId);
+                        foreach (SubscriptionInfo subInfo in subscriptionList)
+                        {
+                            subs += subInfo.SubscriptionId + ",";
+                        }
                     }
+                }
 
-                    //TODO: Find a way to get userId that is being used to retrive subscriptions
-                    //Currently if there occurs a situation, the subscription list retrieved does not match the supplied subscriptionIds
-                    //There is not way to let the user know as to for what user the subscription list was retrieved.
-                    string exceptionString = string.Format("Either no subscription was provided in connection string (e.g. TEST_CSM_ORGID_AUTHENTICATION=SubscriptionId=<subscritionId> or " +
-                        "The provided SubscriptionId in connections string - '{0}' does not match the list of subscriptions associated with the account \r\n {1}", this.SubscriptionId, sb.ToString());
+                if (string.IsNullOrEmpty(matchedSubscriptionId))
+                {
+                    sb.AppendLine(string.Format("SubscriptionList:'{0}' retrieved for the user/spn id '{1}', do not match with the provided subscriptionId '{2}' in connection string",
+                        subs, callerId, this.SubscriptionId));
 
-                    throw new Exception(exceptionString);
+                    throw new Exception(sb.ToString());
+                }
+            }
+            else
```

Comment:
So, previous behavior was to allow not settign the subscription if only one subscription was available in the list subscriptiosn call.  This makes it easier for the scenario where people are using test credentials that are only authorized for a single subscription, which is a common scenario.